go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -3838,13 +3838,280 @@ var DoltAutoIncrementTests = []queries.ScriptTest{
 			},
 		},
 	},
-}
-
-var BrokenAutoIncrementTests = []queries.ScriptTest{
 	{
-		// truncate table doesn't reset the persisted auto increment counter of tables on other branches, which leads to
-		// the value not resetting to 1 after a truncate if the table exists on other branches, even if truncated on every
-		// branch
+		Name: "delete all rows in table",
+		SetUpScript: []string{
+			"create table t (a int primary key auto_increment, b int)",
+			"call dolt_add('.')",
+			"call dolt_commit('-am', 'empty table')",
+			"insert into t (b) values (1), (2)",
+			"call dolt_commit('-am', 'two values on main')",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "delete from t",
+				Expected: []sql.Row{{types.NewOkResult(2)}},
+			},
+			{
+				Query:            "alter table t auto_increment = 1",
+				SkipResultsCheck: true,
+			},
+			{
+				// empty tables, start at 1
+				Query:    "insert into t (b) values (7), (8)",
+				Expected: []sql.Row{{types.OkResult{RowsAffected: 2, InsertID: 1}}},
+			},
+			{
+				Query: "select * from t order by a",
+				Expected: []sql.Row{
+					{1, 7},
+					{2, 8},
+				},
+			},
+		},
+	},
+	{
+		Name: "set auto-increment below current max value",
+		SetUpScript: []string{
+			"create table t (a int primary key auto_increment, b int)",
+			"call dolt_add('.')",
+			"call dolt_commit('-am', 'empty table')",
+			"insert into t (b) values (1), (2), (3), (4)",
+			"call dolt_commit('-am', 'two values on main')",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:            "alter table t auto_increment = 2",
+				SkipResultsCheck: true,
```

Comment:
(minor) should we just expect this to be a Query OK response with no rows updated, like MySQL returns? Might be nice just to assert that there isn't an error and use less `SkipResultsChecks`.

Reply:
SkipResultsCheck still asserts no error, it just doesn't inspect the return rows